### PR TITLE
Issue #20282:combine vendor and vendorworkbench

### PR DESCRIPTION
**Pull request by garyhgohoos**

### CODE REVIEW

File: widgets/vendorcluster.cpp [hunk 1]
```diff
@@ -29,26 +49,239 @@ VendorLineEdit::VendorLineEdit(QWidget *pParent, const char *pName) :
...
+
+  emit editable(p);
+  return p;
+}
+
+void VendorLineEdit::sParse()
+{
+// TODO - don't understand why
+//  if (_editMode)
+//    return;
```
Comment (gilmoskowitz) [inline widgets/vendorcluster.cpp]: Re: `TODO` - could it be that `return` here allows entering a previously unused vendor number?

File: widgets/vendorcluster.h [hunk 2]
```diff
@@ -11,29 +11,107 @@
...
+class QDropEvent;
+class QMouseEvent;
+
+//class VendInfo;
+
+#define __allVendors    0x01
+#define __activeVendorss 0x02
+
+#define CRMACCT_ID      6
+#define ISVENDOR        7
```
Comment (gilmoskowitz) [inline widgets/vendorcluster.h]: These are pretty far removed from where they're used and they only appear to be used once each. There's got to be a better way.

File: widgets/vendorcluster.cpp [hunk 3]
```diff
@@ -29,26 +49,239 @@ VendorLineEdit::VendorLineEdit(QWidget *pParent, const char *pName) :
...
+  if (model() && _id != -1)
+  {
+    setUiName("vendor");
+    setEditPriv("MaintainVendors");
+    setViewPriv("ViewVendors");
+    setNewPriv("MaintainVendors");
+    _idColName="vend_id";
+    sUpdateMenu();
+
+    _crmacctId = model()->data(model()->index(0,CRMACCT_ID)).toInt();
```
Comment (gilmoskowitz) [inline widgets/vendorcluster.cpp]: There should be a way to derive the column number from the model and the column _name_. Using hard-coded constants like this is fragile and caused us trouble with characteristics handling.

Comment (garyhgohoos) [inline widgets/vendorcluster.cpp]: used custCluster as a template.  don't know how to improve.

**Pull request merged**